pabi_procurement/models/purchase_requisition.py

```diff
@@ -57,27 +67,59 @@ class PurchaseRequisition(models.Model):
     @api.model
     def create(self, vals):
         create_rec = super(PurchaseRequisition, self).create(vals)
-        sum_total = 0.0
+        AccTax = self.env['account.tax']
+        total_untaxed = 0.0
+        tax_amount = 0.0
         if 'line_ids' in vals:
             if len(vals['line_ids']) > 0:
                 for line_rec in vals['line_ids']:
-                    line_flds = line_rec[2]
-                    sum_total += \
-                        line_flds['product_qty'] * line_flds['price_unit']
-            create_rec.amount_total = sum_total
+                    field = line_rec[2]
+                    for rec_taxes in field['taxes_id']:
+                        for rec_tax in rec_taxes[2]:
+                            domain = [('id', '=', rec_tax)]
+                            found_tax = AccTax.search(domain)
+                            if found_tax.type == 'percent':
+                                tax_amount += field['product_qty'] * (
+                                    field['price_unit'] * found_tax.amount
+                                )
+                            elif found_tax.type == 'fixed':
+                                tax_amount += field['product_qty'] * (
+                                    field['price_unit'] + found_tax.amount
+                                )
+                    untaxed = field['product_qty'] * field['price_unit']
+                    total_untaxed += untaxed
+            create_rec.amount_untaxed = total_untaxed
+            create_rec.amount_tax = tax_amount
+            create_rec.amount_total = total_untaxed + tax_amount
         return create_rec
 
     @api.multi
     def write(self, vals):
         super(PurchaseRequisition, self).write(vals)
-        prql_obj = self.env['purchase.requisition.line']
+        PRLine = self.env['purchase.requisition.line']
         sum_total = 0.0
+        total_untaxed = 0.0
+        tax_amount = 0.0
         domain = [('requisition_id', '=', self.id)]
-        found_recs = prql_obj.search(domain)
+        found_recs = PRLine.search(domain)
         for rec in found_recs:
-            sum_total += rec.product_qty * rec.price_unit
+            for rec_tax in rec.taxes_id:
+                if rec_tax.type == 'percent':
+                    tax_amount += rec.product_qty * (
+                        rec.price_unit * rec_tax.amount
+                    )
+                elif rec_tax.type == 'fixed':
+                    tax_amount += rec.product_qty * (
+                        rec.price_unit + rec_tax.amount
+                    )
+            total_untaxed += rec.product_qty * rec.price_unit
+            sum_total += rec.price_subtotal
         edited = super(PurchaseRequisition, self).\
```

Comment:
super(PR, self).write(..) is called 2 times. Any reason?
Possible to calculate then update to vals, then call super(PR, self).write(..) just once?
For example, you can,

> vals.update({
>     'x': 1,
>     'y': 2,
>     'z': 3
> })
> super(PurchaseRequisition, self).write(vals)